src/basic/confidential-virt.c

```diff
@@ -209,6 +209,23 @@ static ConfidentialVirtualization detect_confidential_virtualization_impl(void)
 
         return CONFIDENTIAL_VIRTUALIZATION_NONE;
 }
+#elif defined(__s390x__)
+static ConfidentialVirtualization detect_confidential_virtualization_impl(void) {
+        int fd;
+        char c;
+
+        fd = open("/sys/firmware/uv/prot_virt_guest", O_RDONLY);
```

Comment:
please use the `read_full_virtual_file()` helper

Reply:
Done